e2e-tests/ukef/cypress/e2e/journeys/gef/submit-AIN-deal-with-issued-facilities-deal-cancelled-maker-unable-to-view-update-issue-facility-section/submit-AIN-deal-with-issued-facilities-deal-cancelled-maker-unable-to-view-update-issue-facility-section.spec.js

```diff
@@ -0,0 +1,111 @@
+
+      cy.createGefFacilities(dealId, [anIssuedCashFacility()], BANK1_MAKER1).then((createdFacilities) => {
+        dealFacilities.push(createdFacilities.details);
+      });
+    });
+  });
+
+  beforeEach(() => {
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+  });
+
+  after(() => {
+    cy.clearCookies();
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+  });
+
+  it('Gef AIN deal with issued facilities is submitted to UKEF, user cancel deal in TFM. Maker unable to view update issue facility section on portal', () => {
+    //---------------------------------------------------------------
+    // portal maker submits gef deal for review
+    //---------------------------------------------------------------
+    cy.login(BANK1_MAKER1);
+    gefPages.applicationDetails.visit(deal);
+    submitButton().click();
+    cy.url().should('eq', relative(`/gef/application-details/${dealId}/submit`));
+    cy.keyboardInput(gefPages.applicationSubmission.commentsField(), 'go');
+    submitButton().click();
+
+    //---------------------------------------------------------------
+    // portal checker submits gef deal to ukef
+    //---------------------------------------------------------------
+    cy.login(BANK1_CHECKER1);
+    gefPages.applicationDetails.visit(deal);
+    submitButton().click();
+    gefPages.applicationSubmission.confirmSubmissionCheckbox().check();
+    submitButton().click();
+
+    // expect to land on the /submit-to-ukef page with a success message
+    cy.url().should('include', '/submit-to-ukef');
+
+    //---------------------------------------------------------------
+    // user login to TFM and cancel the gef deal
+    //----------------------------------------------------------------
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+    cy.forceVisit(TFM_URL);
+
+    cy.tfmLogin(PIM_USER_1);
+    cy.forceVisit(`${TFM_URL}/case/${dealId}/deal`);
+    tfmPages.caseDealPage.cancelDealButton().click();
+    cy.keyboardInput(tfmPages.reasonForCancellingPage.reasonForCancellingTextBox(), 'cancel');
+    tfmPartials.continueButton().click();
+    cy.url().should('include', '/bank-request-date');
+    cy.completeDateFormFields({ idPrefix: 'bank-request-date' });
+    tfmPartials.continueButton().click();
+
+    cy.url().should('include', '/effective-from-date');
+    cy.completeDateFormFields({ idPrefix: 'effective-from-date' });
+    tfmPartials.continueButton().click();
+
+    cy.url().should('include', '/check-details');
+    tfmPages.checkDetails.dealDeletionButton().click();
+
+    cy.url().should('include', '/deal');
+    tfmPages.caseDealPage
+      .dealStage()
+      .invoke('text')
+      .then((text) => {
+        expect(text.trim()).to.contain('Cancelled');
+      });
```

Comment:
Can we use `cy.assertText` instead of contains please, so we have strict checking (and less LOC) i.e:

```js
cy.asserText(theSelector(), 'Some text');
```

Reply:
I decided not to test success banner as It is not part of this ticket